### PR TITLE
Added Legend Of Intensity

### PR DESCRIPTION
**Pull request by OssamaRafique**
**Example 1, Custom Dom Element Outside Of Graph HTML Node**
image
**Example 2 Inside Graph Dom Node**
image
Adding Intensity legend is optional. If you want to add it, the simplest way is to provide the data. The data should be in this format.
`[
{
  color: '#fff',
  intensity: 0.1,
  label: '-2'
}
  ]`.

You can use existing setInput method to set the legendData along with heatmap data. like `this.plot.setInput(data, intensityLegendData);` With this option, by default intensity legend will render at bottom of the graph.

You can always change the position using `setIntensityLegendOptions("top" | "bottom" | "left" | "right");`.

If you want the legend to be rendered somewhere else in the dom. You have to provide the selector or dom element along with the position / type of intensity legend you want.
e.g
`setIntensityLegendOptions("top", ".intensity-legend")`

You can also specify width and height.
`setIntensityLegendOptions("top", ".intensity-legend", 400, 500)`

### CODE REVIEW

**Comment by jkanche** [comment]
awesome. I think it would look nice if the height of the gradient bar is within the visualization part of the canvas (exclude the labels)

**Comment by OssamaRafique** [comment]
> awesome. I think it would look nice if the height of the gradient bar is within the visualization part of the canvas (exclude the labels)

Sure, On it.

**Comment by zoecastillo** [comment]
I think the legend height should be settable from the outside. In case of a large heatmap, having the legend be the same height or width would look very odd. So there should be a maximum height that a user can define (or if not given using the same height/width as the heatmap is fine)

**Comment by OssamaRafique** [comment]
> I think the legend height should be settable from the outside. In case of a large heatmap, having the legend be the same height or width would look very odd. So there should be a maximum height that a user can define (or if not given using the same height/width as the heatmap is fine)

Makes sense, Will add that.

**Comment by jkanche** [comment]
@OssamaRafique: you should now be able to tag @zoecastillo as a reviewer on the PRs (Once Janina accepts the collaborator invite)

**Comment by zoecastillo** [comment]
Thank you! This looks good to me.

File: src/BaseGL.js [hunk 1]
```diff
@@ -145,9 +148,13 @@ class BaseGL {
    * @param {Array} data.y, y coordinates
    * @param {Array} data.xlabels, labels along the x-axis
    * @param {Array} data.ylabels, labels along the y-axis
+   * @param {Array} legendData, data for the legend
```
Comment (jkanche) [inline src/BaseGL.js]: legendData is an `object`

Comment (jkanche) [reply src/BaseGL.js]: I am also debating if this should be part of the encodings and not input

**Comment by jkanche** [comment]
couple of things 

```
rollup v1.32.1
bundles src/index.js → dist...
(!) Import of non-existent exports
src/BaseGL.js
3:   isObject,
4:   getMinMax,
5:   hexToRGB,
     ^
6:   rgbToHex,
7:   mixWithWhite,
...and 4 other occurrences
(!) Plugin off-main-thread: Very few browsers support ES modules in Workers. If you want to your code to run in all browsers, set `output.format = "amd";`
created dist in 416ms
bundles src/index.js → dist/ehgl.js...
(!) Import of non-existent exports
src/BaseGL.js
3:   isObject,
4:   getMinMax,
5:   hexToRGB,
     ^
6:   rgbToHex,
7:   mixWithWhite,
...and 4 other occurrences
created dist/ehgl.js in 19ms
```

**Comment by jkanche** [comment]
can you also add a snippet to the README on how one would set up the legend?

File: src/BaseGL.js (hunk 1 above)
Comment (OssamaRafique) [reply src/BaseGL.js]: > object

Actually I made legend data an array of objects. Each object contains label, intensity and color.

Comment (OssamaRafique) [reply src/BaseGL.js]: > I am also debating if this should be part of the encodings and not input

Makes sense.

**Comment by OssamaRafique** [comment]
> can you also add a snippet to the README on how one would set up the legend?

Added. Please check

**Comment by jkanche** [comment]
looks good. lets bump the library version

**Pull request merged**